src/engine/enginebuffer.h

```diff
@@ -77,11 +79,17 @@ class EngineBuffer : public EngineObject {
 
     // This enum is also used in mixxx.cfg
     // Don't remove or swap values to keep backward compatibility
-    enum KeylockEngine {
-        SOUNDTOUCH,
-        RUBBERBAND_FASTER,
-        RUBBERBAND_FINER,
-        KEYLOCK_ENGINE_COUNT,
+    enum class KeylockEngine : std::uint8_t {
+        SoundTouch,
+        RubberBandFaster,
+        RubberBandFiner,
+    };
+
+    // intended for iteration over the KeylockEngine enum
+    constexpr static std::initializer_list<KeylockEngine> kKeylockEngines = {
+            KeylockEngine::SoundTouch,
+            KeylockEngine::RubberBandFaster,
+            KeylockEngine::RubberBandFiner,
     };
```

Comment:
This duplication is a source of later error. 

I am not against using an enum class here. But just because this is a good example how software paradigms and maintainability/readability are not always the same, let me share my view: 

enum classes have been introduced for two reasons
* A protected namespace 
* type-safty 

Part of the type safety is that the value representation can be random. There is no iterator, no compare operator and no ++ operator or implicit numeric value cast because of that reason, this protects the developer from doing wrong. 

However, in our case we need exactly this. The enum is in fact just a util to not deal with the magic numbers from the mixxx.cfg file in our source code. 

We need control over the value, we need the integer presentation and an iterator, the sings that have been removed from the enum class by design.  So from the type-safty aspect an plain old Enum is a better fit ... but without the protected namespace we reintroduce a "strong" issue. 

So I propose to reintroduce just the value KeylockEngine::Count and the integer loop in this case.  

The alternative would be to add the missing operators to this enum class https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#Renum-oper or introduce a generic solution like shown here: https://stackoverflow.com/a/8498694

Reply:
Your review is reasonable and I have a couple notes:

> So I propose to reintroduce just the value KeylockEngine::Count and the integer loop in this case.

I strongly object to this. As you said yourself, we can't be sure about the underlying value. Adding a last entry and assuming it indicates how many constants the enum contains is IMO a bad code smell. The current approach is very careful to not cause undefined behavior. As such, casting from some int to KeylockEngine was carefully avoided. We can't assume that the enum values start at 0 and are consecutive. For example if we ever removed Soundtouch, we can't remove the Soundtouch entry from the enum. We'd also have to modify the loop that loop that does the static cast hack, etc. All of which could cause bugs and UB and from what I can tell, the compiler can't warn us about.

So we need an iterator, if I were to define an iterator on the Keylock enum, the result would essentially be another giant switch-case mess, because due to the reasons I mentioned above, `static_cast<KeylockEngine>(++static_cast<int>(engine))` won't work. 

The problem with the value representation being random is also not a problem when provide explicit values (as suggested in my own comment above). 

My initial plan was to make `kKeylockEngines` be `kAvailableKeylockEngines` and then refactor the other querying methods in terms of that, but that either resulted in unnecessary interdependencies or complex code that would be objected to.